main/githooks.py

```diff
@@ -773,6 +781,24 @@ def check_commit_msg(message, files):
                 return 1
 
     return 0
+jira_id_pattern = re.compile(r'\b[A-Z]{2,5}-[0-9]{1,4}\b')
+
+
+class TestJiraIDPattern(unittest.TestCase):
+    def test_various_strings(self):
+        def _test(input, is_jira=True):
+            m = jira_id_pattern.search(input)
+            self.assertEqual(bool(m), is_jira)
+        _test('BLD-5704')
+        _test("BLD-1234 fixed some builds")
+        _test("fixed some builds BLD-1234 ")
+        _test("fixed some builds (Jira BLD-1234)")
+        _test("fixed some builds\n some more text BLD-1234")
+        _test('lower-1234', False)
+        _test('A-1234', False)
+        _test('ABCDEF-1234', False)
+        _test('BLD-12345', False)
```

Comment:
I suggest the upper limit on number of issues would be better as six digits - mercury is already at 6000 issues.

Reply:
Also, one project at least has a very long initial ID, even though this was a mistake we need to support it.  I will also suggest adding a positive test for "SKETCHER-1".

Reply:
We will update the pattern to [A-Z]{2,8}-[0-9]{1,5}